storage/dsadapter/go.mod

```diff
@@ -0,0 +1,5 @@
+module github.com/ipld/go-ipld-prime/storage/dsadapter
```

Comment:
Why an extra module? If the reason is that you don't want ipld-prime to depend on go-datastore, then perhaps put this code in go-datastore.

If I want to use ipld-prime with datastore, I want to find the wrapper solution in either module - not have to discover an extra module. The IPLD/IPFS ecosystem already has enough Go modules as it is, too :)

Reply:
I just read the README, which confirms my suspicion. I still think this code belongs in either of the two modules.

Reply:
:thinking:  I guess I can't really defend this, other than I think it's going to be operationally easier for me to deal with it here, and I kind of like the idea of having a few other things also in-repo here as sibling packages, so I can say "look in `storage/*` for implementations" in the docs.

(I have another freshly written filesystem-based storage system to add in a sibling directory, shortly, which would make for three of them here, including this one.)

You're right that this code could technically go in the go-datastore repo and module, due to the the success of the key design goal here of not having the new API require any types outside of the standard library.

Reply:
Alright, if someone wants to push this into `go-datastore` be my guest, but I want to see the PR landed in <24 hours, and then you also get to deal with whatever else comes up as I try to add benchmarks in the coming days that reach across old and new code :upside_down_face: Sounds like the road of greater hassle to me, but I dunno.  :shrug: 

Otherwise, I'd suggest we keep it here for now, and revisit in the future if it causes pain or discovery failure.  I don't think it should be difficult to move (by copying to a new home, and eventually removing the older location, after a transition period), if we find it necessary.  Leaving it here is a very "type 2 / revisitable" decision, in other words.